### ISSUE TITLE
Add option to disable class-methods-use-this on inheritance

### ISSUE THREAD

**Issue by IGassmann** [eslint/eslint#12390]
**What rule do you want to change?** `class-methods-use-this`

**Does this change cause the rule to produce more or fewer warnings?** Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?** New option: `disableOnInheritance`.

**Please provide some example code that this change will affect:**

```js
class ConfigService extends ConfigManager {
  provideConfigSpec() {
    return {
      ENV_VAR: 'test';
    };
  }
}
```

**What does the rule currently do for this code?** It throws the error because of the base class method `provideConfigSpec` that doesn't use `this.

**What will the rule do after it's changed?** Will disable the error when inheriting.

**Are you willing to submit a pull request to implement this change?** With some guidance, yes.

**Comment by mdjermanovic** [comment]
Hi @IGassmann, thanks for the proposal!

Could the existing `exceptMethods` option help, it's designed for these situations:

> The exceptMethods option allows you to pass an array of method names for which you would like to ignore warnings. For example, you might have a spec from an external library that requires you to overwrite a method as a regular function (and not as a static method) and does not use this inside the function body. In this case, you can add that method to ignore in the warnings.

It might be a more useful solution than disabling the rule on all methods in all derived classes just for a couple of exceptions.

**Comment by IGassmann** [comment]
The issue with `exceptMethods` is that you need to keep adding new methods. Sometimes, you're using a framework that is using base classes with implementable methods to offer functionality. In this case, it would be nice to have `disableOnInheritance`, so you don't need to add new methods every time you use a new object from this framework.

**Comment by IGassmann** [comment]
Unfortunately not, because rarely a framework would use a base name that could be caught by the regex into the method. At least, I have never stumbled upon one that does that.

BTW, the framework I'm mostly referring is NestJS. But I believe it also applies to Angular.

**Comment by mdjermanovic** [comment]
Thanks for the info! 

The enhancement is open for evaluation, it needs a champion and 3 other :+1:  from the team to be accepted. 

The option seems useful to avoid a long list in `exceptMethods`, but on the other hand looks like it would disable too much, i.e. limit the rule to check just base classes.
